packages/auth/src/authClients/firebase.ts

```diff
@@ -50,82 +74,84 @@ export const firebase = (client: Firebase): AuthClient => {
     }
     return provider
   }
-  // Firebase auth functions return a goog.Promise which as of 2021-05-12 does
-  // not appear to work with try {await} catch blocks as exceptions are not caught.
-  // This client returns a new standard Promise so that the exceptions can be
-  // caught and no changes are required in common auth code located in AuthProvider.tsx
-  const repackagePromise = (
-    fireBasePromise: Promise<any | string>
-  ): Promise<any | string> => {
-    return new Promise((resolve, reject) => {
-      fireBasePromise
-        .then((result) => resolve(result))
-        .catch((error) => reject(error))
-    })
+
+  const loginWithEmailLink = ({ email, emailLink }: Options) => {
+    // dispatch to signIn step1 based on if no emailLink has been provided
+    if (!emailLink) {
+      return sendSignInLinkToEmail(
+        auth,
+        email as string,
+        actionCodeSettings as ActionCodeSettings
+      )
+    }
+    // otherwise validate emailLink
+    if (!isSignInWithEmailLink(auth, emailLink)) {
+      throw new Error('Login failed, invalid email link')
+    }
+
+    return signInWithEmailLink(auth, email as string, emailLink as string)
   }
 
+  // restoreAuthState?(): void | Promise<any>
   return {
     type: 'firebase',
-    client,
-    restoreAuthState: () => repackagePromise(client.auth().getRedirectResult()),
-    login: (
+    client: auth,
+    //restoreAuthState -- perhaps use refreshtoken to reauth?
```

Comment:
This looks like its going to break potentially

Reply:
Ahh yes, good catch.  I'll see about testing this with a local copy of the auth playground and make sure I didn't break the other auth flows and fix it accordingly.

Reply:
@dac09 do you know what the expectation is in regard to what an authclient's restoreAuthState() method is supposed to do? 

Anyhow, I don't think the old implementation's calling getRedirectResult() was correct, having no  implementation made no difference in the behavior for playground-auth's firebase example.  But I've gone ahead and reimplemented it to call currentuser.reload() assuming that is better mapped to the intended purpose of restoreAuthState()

Whether you'd even want to use getRedirectResult() is dependent on which auth flow is being used, and may make more sense to be called by the app because it wouldn't make sense for popup or email link auth flows for example.